backend/debug/inlined_frame_ranges.ml

```diff
@@ -0,0 +1,189 @@
+(******************************************************************************
+ *                             flambda-backend                                *
+ *                       Mark Shinwell, Jane Street                           *
+ * -------------------------------------------------------------------------- *
+ *                               MIT License                                  *
+ *                                                                            *
+ * Copyright (c) 2024 Jane Street Group LLC                                   *
+ * opensource-contacts@janestreet.com                                         *
+ *                                                                            *
+ * Permission is hereby granted, free of charge, to any person obtaining a    *
+ * copy of this software and associated documentation files (the "Software"), *
+ * to deal in the Software without restriction, including without limitation  *
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,   *
+ * and/or sell copies of the Software, and to permit persons to whom the      *
+ * Software is furnished to do so, subject to the following conditions:       *
+ *                                                                            *
+ * The above copyright notice and this permission notice shall be included    *
+ * in all copies or substantial portions of the Software.                     *
+ *                                                                            *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR *
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,   *
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL    *
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER *
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING    *
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER        *
+ * DEALINGS IN THE SOFTWARE.                                                  *
+ ******************************************************************************)
+
+module L = Linear
+
+(* CR mshinwell/xclerc/poechsel:
+
+   1. maybe "set" isn't the correct algebraic structure for the Compute_ranges
+   interface
+
+   2. maybe a list-based structure could be used here instead (not for variable
+   ranges) *)
+
+module Inlined_frames = struct
+  module Key = struct
+    module T0 = struct
+      type t = Debuginfo.t
+
+      let print = Debuginfo.print_compact_extended
+
+      let compare t1 t2 =
+        (* It needs to be the case that [compare shorter longer] returns -1 (and
+           vice-versa) when [shorter] has fewer frames than [longer]. See
```

Comment:
Should the property hold iff
`shorter` is a prefix of `longer`?

Reply:
I don't think it's "if and only if", but I've tried to clarify the comment.

Reply:
(`Dwarf_inlined_frames` is, in fact, only interested in the order when it is known one is a prefix of the other)